lib/network/http_middlewares_test.go

```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"boscoin.io/sebak/lib/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRecoverMiddleware(t *testing.T) {
+	endpoint, err := common.NewEndpointFromString(
+		fmt.Sprintf("http://localhost:%s", getPort()),
+	)
+	require.Nil(t, err)
+
+	network, err := makeTestHTTP2NetworkForTLS(endpoint)
```

Comment:
I suggest another approach for testing. Please check https://github.com/bosnet/sebak/blob/master/lib/network/api/test.go#L21 , this shows http handler test without creating HTTP2 server .

Reply:
That is better?  I thought testing with `Network.AddMiddleware` too. but if that is better, I will change it :-)

Reply:
https://github.com/bosnet/sebak/blob/master/lib/network/api/test.go#L21 is just creating `http test.Server` insteads of `network.Network`  in fact. :->